infrastructure/modules/databases/main.tf

```
@@ -18,25 +25,33 @@ resource "random_password" "kratos_secrets_cookie" {
   upper   = true
 }
 
-resource "postgresql_role" "kratos" {
-  name     = "kratos_user"
+resource "postgresql_role" "service" {
+  for_each = toset(local.databases)
+  name     = "${each.value}_user"
   login    = true
-  password = random_password.kratos_postgresql_password.result
+  password = random_password.postgresql_password[each.value].result
 }
 
 # we are using the postgresql provider here because azurerm_postgresql_database
 # doesn't let us set the owner to a role other than the server admin.
-resource "postgresql_database" "kratos" {
-  name              = "kratos"
-  owner             = postgresql_role.kratos.name
+resource "postgresql_database" "service" {
+  for_each          = toset(local.databases)
+  name              = each.value
+  owner             = postgresql_role.service[each.value].name
   lc_collate        = "C"
   connection_limit  = -1
   allow_connections = true
 }
 
-resource "kubernetes_namespace" "kratos" {
+resource "postgresql_extension" "uuid-ossp" {
+  name     = "uuid-ossp"
+  database = "workspace-service"
+}
+
+resource "kubernetes_namespace" "db" {
```

Let's name the resource `service` as well.
```suggestion
resource "kubernetes_namespace" "service" {
```